### ISSUE TITLE
pubsub subscribe/publish not working

### ISSUE THREAD

**Issue by onichandame** [ipfs/js-ipfs#3282]
- **0.50.2**:

- **Firefox 80.0.1**:

#### Description:

I tried to publish a message using `pubsub.publish` and subscribed to receive the message on the same instance. The publish and subscribe actions were resolved successfully but the callback registered with `pubsub.subscribe` was not triggered.

I have tried the same with version `0.49.0` and `ipfs-http-client` 47.0.1. No problem was seen.

#### Steps to reproduce the error:

```typescript
// compiled by React and run in browser
import { create } from 'ipfs'

;(async () => {
  const encoder = new TextEncoder()
  const decoder = new TextDecoder()
  const ipfs = await create()
  // subscribe to topic
  await ipfs.pubsub.subscribe('topic', message => {
    // This callback is never called

    // display the payload on receiving the message
    alert(decoder.decode(message.data))
  })
  // publish message to topic
  await ipfs.pubsub.publish('topic', encoder.encode(`hi bro`))
)()

**Comment by vasco-santos** [comment]
Hey @onichandame 
Thanks for reporting this. I did this test on a new branch in a example repo I have created, but I am not able to reproduce it in firefox 80.0.1 with `ipfs@0.50.2`.
Can you try my [example](https://github.com/vasco-santos/ipfs-browser-pubsub-example/blob/chore/same-peer/index.js) in your browser?
I don't see any differences to your example. Can you provide me an example already setup that I can run?

**Comment by onichandame** [comment]
Thanks for your example leading me to the root cause.

It was my fault passing a string array(e.g. `["topic"]`) to the `subscribe` function as the topic, instead of a string(e.g. `"topic"`), that caused the error. Wierdly the old version and the http client had no problem about it.

**Comment by vasco-santos** [comment]
Thanks for letting me know that @onichandame 
We did support providing an array of topics on the pubsub router layer (eg. https://github.com/ChainSafe/js-libp2p-gossipsub), but we removed support for that in the latest release as it can be easily done in the application layer if needed. However, from `ipfs` and `libp2p` we did not document that possibility and I did not expect that you were using it.
